tests/expectations/nested_nonzero.tag.c

```diff
@@ -0,0 +1,13 @@
+#include <stdarg.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+typedef uint32_t Handle_File;
+
+struct Node {
+  Handle_File file;
+  uint32_t maybe_file;
```

Comment:
like here.  Ideally both fields would be `Handle_File`.

Reply:
Yeah, agreed... In practice it shouldn't be a huge deal for C. For C++ it can be a bit more annoying (I can imagine someone defining their own `Handle<>` for example.

Reply:
The latest patch fixes this, at the expense of the `infinite_recursion_typedef_monomorph` test going into an infinite loop.  I'd really love to be able to use the `Monomorphs` data here, but that's not generated until later in the process, and it looks like it depends on all of the types already having been simplified.